app/upgrades/v2/upgrades.go

```diff
@@ -15,9 +29,59 @@ func CreateUpgradeHandler(mm *module.Manager,
 	configurator module.Configurator,
 	_ upgrades.BaseAppParamManager,
 	keepers *keepers.AppKeepers) upgradetypes.UpgradeHandler {
+	for _, subspace := range keepers.ParamsKeeper.GetSubspaces() {
+		subspace := subspace
+
+		// ctx.Logger().Info(fmt.Sprintf("subspace.Name:%s", subspace.Name()))
+		// ctx.Logger().Info(fmt.Sprintf("i:%d", i))
+		var keyTable paramstypes.KeyTable
+		switch subspace.Name() {
+		case authtypes.ModuleName:
+			keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+		case banktypes.ModuleName:
+			keyTable = banktypes.ParamKeyTable() //nolint:staticcheck
+		case stakingtypes.ModuleName:
+			keyTable = stakingtypes.ParamKeyTable() //nolint:staticcheck
+		case minttypes.ModuleName:
+			keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+		case distrtypes.ModuleName:
+			keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck
+		case slashingtypes.ModuleName:
+			keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+		case govtypes.ModuleName:
+			keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+		case crisistypes.ModuleName:
+			keyTable = crisistypes.ParamKeyTable() //nolint:staticcheck
+		case wasmtypes.ModuleName:
+			keyTable = wasmtypes.ParamKeyTable() //nolint:staticcheck
+		}
+
+		if !subspace.HasKeyTable() {
+			subspace.WithKeyTable(keyTable)
+		}
```

Comment:
I haven't seen similar upgrade codebase before, will need to check if it's working fine on actual upgrade process.

Reply:
it's work and cosmos sdk require this upgrade process.
refs
https://github.com/cosmos/cosmos-sdk/blob/v0.47.1/simapp/upgrades.go
https://github.com/CosmWasm/wasmd/blob/main/app/upgrades.go

Reply:
I see - thanks for your reference.